### PR TITLE
Configure uwsgi_exporter sidecar container, gated behind a per-instance toggle.

### PR DESCRIPTION
**Pull request by EvanKrall**
for PAASTA-17092

### CODE REVIEW

File: paasta_tools/kubernetes_tools.py [hunk 1]
```diff
@@ -724,6 +712,16 @@ def get_sidecar_containers(
         service_namespace_config: ServiceNamespaceConfig,
         hacheck_sidecar_volumes: Sequence[DockerVolume],
     ) -> Sequence[V1Container]:
+        return self.get_hacheck_sidecar_container(
+            system_paasta_config, service_namespace_config, hacheck_sidecar_volumes,
+        ) + self.get_uwsgi_exporter_sidecar_container(system_paasta_config)
```
Comment (stug) [inline paasta_tools/kubernetes_tools.py:717]: nit, but I find returning 1-element lists a little unexpected

File: paasta_tools/utils.py [hunk 2]
```diff
@@ -2556,6 +2558,16 @@ def get_legacy_autoscaling_signalflow(self) -> str:
...
 
+    def get_uwsgi_exporter_sidecar_image_url(self) -> str:
+        """Get the docker image URL for the uwsgi_exporter sidecar container"""
+        return self.config_dict.get(
+            "uwsgi_exporter_sidecar_image_url",
+            "docker-paasta.yelpcorp.com:443/uwsgi_exporter-k8s-sidecar:v1.0.0",
+        )
+
+    def default_should_run_uwsgi_exporter_sidecar(self) -> bool:
+        return self.config_dict.get("default_should_run_uwsgi_exporter_sidecar", False,)
```
Comment (stug) [inline paasta_tools/utils.py:2569]: nit: extra comma at the end?  Doesn't look like it changes things, though.

**Review by stug — APPROVED** [review]
Looks roughly sane.  I have small comments but only because I am a curmudgeon.

File: paasta_tools/kubernetes_tools.py (hunk 1 above)
Comment (EvanKrall) [reply paasta_tools/kubernetes_tools.py:717]: Yeah, I guess that's weird. It just saves me from having to do something like 
```
hacheck_sidecar = self.get_hacheck_sidecar_container(...)
if hacheck_sidecar is not None:
    sidecars.append(hacheck_sidecar)
```
maybe there's a better one-liner for that.

File: paasta_tools/utils.py (hunk 2 above)
Comment (EvanKrall) [reply paasta_tools/utils.py:2569]: huh, surprised that Black let that stay.

**Review by qui — APPROVED** [review]

**Comment by EvanKrall** [comment]
@stug I refactored to have those methods return Optional[V1Container] instead of Lists, and fixed the weird trailing comma.

**Review by stug — APPROVED** [review]

**Pull request merged**